lib/utils/helpers.js

```diff
@@ -28,13 +38,107 @@ const formatUserReadStat = (readStats) => {
   return userReadStat;
 };
 
-const getTokenFromReq = req => (
-  req.body.token || req.query.token || req.headers.authorization || req.params.token
-);
+const getTokenFromReq = req => req.body.token
+|| req.query.token
+|| req.headers.authorization
+|| req.params.token;
+
+const articleByAuthorWhere = (value) => {
+  const authorName = value.split(' ');
+  if (authorName.length === 2) {
+    const authorWhere = {
+      [Op.or]: [
+        {
+          firstname: {
+            [Op.iLike]: `%${authorName[0]}%`
+          },
+          lastname: {
+            [Op.iLike]: `%${authorName[1]}%`
+          }
+        },
+        {
+          firstname: {
+            [Op.iLike]: `%${authorName[1]}%`
+          },
+          lastname: {
+            [Op.iLike]: `%${authorName[0]}%`
+          }
+        }
+      ]
+    };
+    return authorWhere;
+  }
+  const authorWhere = {
+    [Op.or]: [
+      {
+        firstname: {
+          [Op.iLike]: `%${authorName}%`
+        }
+      },
+      {
+        lastname: {
+          [Op.iLike]: `%${authorName}%`
+        }
+      }
+    ]
+  };
+  return authorWhere;
```

Comment:
The reason I added the first where clause is to be able to streamline the search result in case the user is searching by both firstname and lastname of the author. If that is not included then the where clause will always be true whenever the firstname or lastname matches the search query.

Reply:
Why are assuming first name and last name based on position? What happens when the user changed the positions?